src/oemof/network/network/entity.py

```diff
@@ -51,7 +51,12 @@ def __init__(self, label, *, custom_properties=None):
         self.custom_properties = custom_properties
 
     def __eq__(self, other):
-        return id(self) == id(other)
+        if isinstance(other, Entity):
+            # We could also check if labels are the same,
+            # but decided on minor change.
+            return id(self) == id(other)
```

Comment:
I have thought about this for a while. This implementation would lead to the following inconsistency:
```python
node1 = Node("my_label")
node2 = Node("my_label")

node1 == "my_label"  # True
node2 == "my_label"  # True
node1 == node2  # False
```
 
Thus, I think we should always compare the labels.

```suggestion
            return self._label == other._label
```